### ISSUE TITLE
iTunes store sign-in required after Signal App update

### ISSUE THREAD

**Issue by mikeweilgart** [signalapp/Signal-iOS#4276]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
After most recent update to Signal App, when attempting to open the app, I am prompted to sign in to the iTunes Store.

### Steps to reproduce
Since I have no way to downgrade the App, I can only list the steps I followed; I cannot confirm reproducibility.

- Update Signal App to the latest version.
- Attempt to open it.

It is worth noting that I also synced music to this iPhone for the first time (from my iTunes) last night (between the last time I opened Signal and this time).  Not sure if that is related.

**Actual result:** The Signal App doesn't start normally; instead the Signal logo is shown (full screen) and a prompt appears, "Sign In to iTunes Store".

**Expected result:** Signal App should start and open normally and be usable, the same way it was before the update.

### Screenshots

N/A (I don't think)

### Device info

**Device**: iPhone 6 Plus

**iOS version**: 12.4.1

**Signal version:** 2.43.3

### Link to debug log

I won't be able to access Signal settings until after I sign in to iTunes Store, which I am delaying in case there is some info the developers would like before I do that.  Please let me know.

**Comment by michaelkirk-signal** [comment]
That's pretty strange. 

Have you recently restored this device? Are you seeing this behavior with any other non-built-in apps?

Before the update, when was the last time you'd successfully launched Signal?

**Comment by mikeweilgart** [comment]
No recent restore, just the music syncing.  (I had trouble syncing the music as iTunes falsely reported how much free space was available, but I'm pretty sure that is entirely unrelated.)

Signal was opened the day before.  The timeline was: Signal was observed to be working; iPhone had music synced to it; Signal app was updated; Signal was observed to require sign in.

Unfortunately I can't answer about if this behavior affected other non-built-in apps, as we use Signal enough that I had to sign in the day after I wrote this bug report.  :)  (This was my wife's iPhone, by the way.)  I don't *think* other apps were affected, but I am not sure.
